src/tests.rs

```diff
@@ -1,279 +1,163 @@
-                export_type: crate::types::ExportType::Json,
-                file_name: "tmp3".to_owned(),
-            },
-        ];
-
-        for (input, expected) in inputs.iter().zip(expected) {
-            assert_eq!(parse_args(input.to_owned()).top_level_arg, expected)
-        }
-    }
-
-    #[test]
-    #[should_panic]
-    fn not_enough_arguments_ii() {
-        parse_args(vec!["opus".to_string(), "add".to_string()]);
-    }
-
-    #[test]
-    #[should_panic]
-    fn not_enough_arguments_export_1() {
-        parse_args(vec!["opus".to_string(), "export".to_string()]);
-    }
-
-    #[test]
-    #[should_panic]
-    fn not_enough_arguments_export_2() {
-        parse_args(vec![
-            "opus".to_string(),
-            "export".to_string(),
-            "csv".to_owned(),
-        ]);
-    }
-
-    #[test]
-    #[should_panic]
-    fn not_enough_arguments_export_3() {
-        parse_args(vec![
-            "opus".to_string(),
-            "export".to_string(),
-            "file".to_owned(),
-        ]);
-    }
+    use crate::cli::{cli_clear, cli_del_task, cli_fin_task, cli_get_tasks};
 
     #[test]
     fn insert_task() {
-        let r = parse_args(vec![
-            "opus".to_string(),
-            "add".to_string(),
-            "update excel sheet #work @today ,,,".to_string(),
-        ]);
-        let task = r.input.task.unwrap();
+        use crate::cli::cli_add_task;
+        use crate::{db::open_db, types::Task};
+
+        let task = Task::from(String::from("update excel sheet #work @today .5"));
         let db = open_db();
+
         db.create_table_if_missing();
         cli_add_task(&db, task);
+
         db.con.close().expect("Closing Database failed.");
     }
 
     #[test]
     fn get_task_by_id() {
-        let r = parse_args(vec![
-            "opus".to_string(),
-            "add".to_string(),
-            "update excel sheet #work @today ,,,".to_string(),
-        ]);
-        let task = r.input.task.unwrap();
+        use crate::cli::cli_add_task;
+        use crate::{db::open_db, types::Task};
+
+        let task = Task::from(String::from("update excel sheet #work @today .5"));
         let db = open_db();
+
         db.create_table_if_missing();
+        cli_add_task(&db, task.clone());
 
-        cli_add_task(&db, task);
-        let tasks = cli_get_tasks(&db, db.con.last_insert_rowid().to_string());
-        let task = tasks.get(0).unwrap();
+        let last_id = db.con.last_insert_rowid();
+
+        let tasks = cli_get_tasks(&db, last_id.to_string(), false);
+        let task1 = tasks.get(0).unwrap();
 
-        assert_eq!(task.title, "update excel sheet");
-        assert_eq!(task.tag, "#work");
-        assert_eq!(task.priority, 3);
+        assert_eq!(task.title, task1.title);
+        assert_eq!(task.tag, task1.tag);
+        assert_eq!(task.priority, task1.priority);
+        assert_eq!(task.due, task1.due);
+        assert_eq!(task.finished, task1.finished);
```

Comment:
Maybe we could define a function e.g. `Task::equals_content` that checks whether the content (everything except the id) is the same to reduce code redundancy since this is used three times.

Because that function would be only used inside the tests module you would receive a compiler warning that it is unused (since you re-enabled dead code warnings :+1:). To fix this you could place the impl block for that function inside the test module or add `#[cfg(test)]` to the function definition.